src/custom_requests.ml

```diff
@@ -14,6 +14,24 @@ let send_request client req params =
 
 let ocamllsp_prefixed s = "ocamllsp/" ^ s
 
+module DocumentPosition = struct
+  type t =
+    { uri : Uri.t
+    ; position : [ `Position of Position.t | `Range of Range.t ]
+    }
+
+  let encode { uri; position } ~pos_name =
+    let open Jsonoo.Encode in
+    let uri = ("uri", string @@ Uri.toString uri ()) in
+    let position =
+      ( pos_name
+      , match position with
+        | `Position p -> Position.json_of_t p
+        | `Range r -> Range.json_of_t r )
+    in
+    [ uri; position ]
```

Comment:
You are right. I think it makes sense to separate them as some of the custom requests return a range (`Type_search`). 
I can do a final commit separating uri from pos

Reply:
Something like

```ocaml

module DocumentPosition = struct
  type t = { loc : [ `Position of Position.t | `Range of Range.t ] }

  let encode ?(pos_name = None) loc =
    let open Jsonoo.Encode in
    let position =
      match loc with
      | `Position p ->
        (Option.value ~default:"position" pos_name, Position.json_of_t p)
      | `Range r ->
        (Option.value ~default:"range" pos_name, Range.json_of_t r)
    in
    [ position ]
end

module DocumentUri = struct
  type t = { uri : Uri.t }

  let encode { uri } =
    let open Jsonoo.Encode in
    let uri = ("uri", string @@ Uri.toString uri ()) in
    [ uri ]
end
```

Reply:
Since we don't have multiple usages right now it's hard to decide which is the best abstraction. I think we should just revert that refactor and re-introduce one when it does make more sense.